pkg/contract_watcher/shared/getter/interface_getter.go

```diff
@@ -32,27 +32,24 @@ type interfaceGetter struct {
 	fetcher.Fetcher
 }
 
-func NewInterfaceGetter(blockChain core.BlockChain) *interfaceGetter {
+// NewInterfaceGetter returns a new InterfaceGetter
+func NewInterfaceGetter(blockChain core.BlockChain) InterfaceGetter {
 	return &interfaceGetter{
 		Fetcher: fetcher.Fetcher{
 			BlockChain: blockChain,
 		},
 	}
 }
 
-// Used to construct a custom ABI based on the results from calling supportsInterface
-func (g *interfaceGetter) GetABI(resolverAddr string, blockNumber int64) (string, error) {
+// GetABI is used to construct a custom ABI based on the results from calling supportsInterface
+func (g *interfaceGetter) GetABI(resolverAddr string, blockNumber int64) string {
 	a := constants.SupportsInterfaceABI
 	args := make([]interface{}, 1)
 	args[0] = constants.MetaSig.Bytes()
 	supports, err := g.getSupportsInterface(a, resolverAddr, blockNumber, args)
-	if err != nil {
-		return "", fmt.Errorf("call to getSupportsInterface failed: %v", err)
+	if err != nil || !supports {
+		return ""
```

Comment:
This if what I'm referring to above. Passing on the error makes it possible to differentiate between the call to the client failing and `!supports`. The empty string makes for a confusing error in the test :)